### PR TITLE
src:  use common functionality for docker cred retrieval

### LINKED ISSUE #751: On-cluster build: Try to get registry credentials from a local developer machine

### PR DESCRIPTION
**Pull request by matejvasek**
resolves #751 #752

### CODE REVIEW

File: pipelines/tekton/pipeplines_provider.go [hunk 1]
```diff
@@ -97,16 +97,21 @@ func (pp *PipelinesProvider) Run(ctx context.Context, f fn.Function) error {
...
+	registry, err := docker.GetRegistry(f.Image)
+	if err != nil {
+		return err
+	}
+
 	_, err = k8s.GetSecret(ctx, getPipelineSecretName(f), pp.namespace)
 	if errors.IsNotFound(err) {
 		pp.progressListener.Stopping()
-		creds, err := pp.containerRegistryCredentialsCallback()
+		creds, err := pp.credentialsProvider(ctx, registry)
```
Comment (zroubalik) [inline pipelines/tekton/pipeplines_provider.go:108]: Have you actually test this? I am not sure that registry == server` in this context. For `docker.io` registry a server `https://index.docker.io/v1/` was needed.

Comment (matejvasek) [reply pipelines/tekton/pipeplines_provider.go:108]: I tested it with quay.io and it worked.

Comment (matejvasek) [reply pipelines/tekton/pipeplines_provider.go:108]: that logic is also used for local push

Comment (matejvasek) [reply pipelines/tekton/pipeplines_provider.go:108]: what happened when you used docker.io?

Comment (matejvasek) [reply pipelines/tekton/pipeplines_provider.go:108]: maybe it's some quirk in `lifecycle` binary? Locally just `docker.io` works fine.

Comment (matejvasek) [reply pipelines/tekton/pipeplines_provider.go:108]: in general `docker.io` is weird and treated as special case :disappointed:

**Comment by matejvasek** [comment]
see https://github.com/google/go-containerregistry/blob/main/pkg/authn/keychain.go#L116-L122

File: pipelines/tekton/pipeplines_provider.go (hunk 1 above)
Comment (matejvasek) [reply pipelines/tekton/pipeplines_provider.go:110]: see https://github.com/google/go-containerregistry/blob/main/pkg/authn/keychain.go#L116-L122

Comment (zroubalik) [reply pipelines/tekton/pipeplines_provider.go:111]: Yeah, it didn't work for `docker.io`.
Cool, let me try that.

**Review by zroubalik — COMMENTED** [review]
/lgtm
/approve

Great, thanks!

**Pull request merged**